From khilman@mvista.com Wed May 28 15:18:58 2008
Date: Tue, 20 May 2008 15:45:46 -0700
From: Kevin Hilman <khilman@mvista.com>
To: linux-rt-users@vger.kernel.org
Cc: linux-omap@vger.kernel.org, Kevin Hilman <khilman@mvista.com>
Subject: [PATCH 2/5] RT: wrap BUILD_CHECK_IRQ_FLAGS

The macros used by BUILD_CHECK_IRQ_FLAGS() are defined in kernel.h,
but this file is sometimes included before kernel.h (or even by
kernel.h)

Signed-off-by: Kevin Hilman <khilman@mvista.com>
---
 include/linux/irqflags.h |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6.26/include/linux/irqflags.h
===================================================================
--- linux-2.6.26.orig/include/linux/irqflags.h
+++ linux-2.6.26/include/linux/irqflags.h
@@ -11,11 +11,15 @@
 #ifndef _LINUX_TRACE_IRQFLAGS_H
 #define _LINUX_TRACE_IRQFLAGS_H
 
+#if !defined(BUILD_CHECK_IRQ_FLAGS) && defined(typecheck)
 #define BUILD_CHECK_IRQ_FLAGS(flags)					\
 	do {								\
 		BUILD_BUG_ON(sizeof(flags) != sizeof(unsigned long));	\
 		typecheck(unsigned long, flags);			\
 	} while (0)
+#else
+#define BUILD_CHECK_IRQ_FLAGS(flags)
+#endif
 
 #ifdef CONFIG_TRACE_IRQFLAGS
   extern void trace_softirqs_on(unsigned long ip);
